File: ipman.shared/Entity/Join/PostWager.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ipman.shared.Entity.Join
{
    public class PostWager: EntityBase
    {
        public Guid PostID { get; set; }
        public Post Post { get; set; }
        public Guid WagerID { get; set; }
        public Wager Wager { get; set; }
        public int Prediction { get; set; }
        public int? RangeLength { get; set; }
    }
}

File: ipman/ClientApp/entity/join/PostWager.ts
import { EntityBase } from "../EntityBase";
import Post from "../Post";
import Wager from "../Wager";

export default interface PostWager extends EntityBase
{
    PostID: string;
    Post: Post;
    WagerID: string;
    Wager: Wager;
    Prediction: number;
    RangeLength: number | null;
}